src/misc/AttachShadowPolyfill.ts

```diff
@@ -0,0 +1,43 @@
+import { $$ } from '../utils/Dom';
+import 'styling/_AttachShadowPolyfill';
+
+export async function attachShadow(element: HTMLElement, init: ShadowRootInit): Promise<ShadowRoot | HTMLElement> {
+  if (element.attachShadow) {
+    return element.attachShadow(init);
+  }
+  const iframe = $$('iframe', { className: 'coveo-shadow-iframe', scrolling: 'no' }).el as HTMLIFrameElement;
+  const onLoad = new Promise(resolve => iframe.addEventListener('load', () => resolve()));
+  element.appendChild(iframe);
+  await onLoad;
+
+  const iframeBody = iframe.contentDocument.body as HTMLBodyElement;
+  iframeBody.style.margin = '0';
+  inheritStyle(iframeBody, document.body, ['fontFamily', 'fontSize', 'fontWeight']);
+  const shadowRoot = $$('div').el;
+  iframeBody.appendChild(shadowRoot);
+  autoUpdateHeight(iframe, shadowRoot);
```

Comment:
While the shadowDOM gets resized like a normal element, iframes have a fixed size and need javascript to be resized based on the size of their content.